src/components/content/order/common/AvailabilityZoneInfo.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import React, { Dispatch, SetStateAction, useMemo } from 'react';
+import { Flex, Form, Radio, RadioChangeEvent } from 'antd';
+import '../../../../styles/service_order.css';
+import { AvailabilityZoneConfig } from '../../../../xpanse-api/generated';
+
+export const AvailabilityZoneInfo = ({
+    availabilityZonesData,
+    availabilityZoneConfigs,
+    selectAvailabilityZones,
+    setSelectAvailabilityZones,
+}: {
+    availabilityZonesData: string[];
+    availabilityZoneConfigs: AvailabilityZoneConfig[] | undefined;
+    selectAvailabilityZones: Record<string, string>;
+    setSelectAvailabilityZones: Dispatch<SetStateAction<Record<string, string>>>;
+}): React.JSX.Element => {
+    const availabilityZones = useMemo<string[]>(() => {
```

Comment:
i think it makes more sense to move this logic to parent component and use only a setting the newly selected AZ here.